singer_sdk/testing/factory.py

```diff
@@ -0,0 +1,237 @@
+"""Test Class Factory."""
+
+from __future__ import annotations
+
+from typing import Any, cast
+
+import pytest
+
+from singer_sdk import Tap, Target
+
+from .runners import TapTestRunner, TargetTestRunner
+from .suites import (
+    tap_stream_attribute_tests,
+    tap_stream_tests,
+    tap_tests,
+    target_tests,
+)
+
+
+def get_test_class(
+    test_runner: TapTestRunner | TargetTestRunner,
+    test_suites: list,
+    suite_config: dict | None,
```

Comment:
@kgpayne Do you think it's worth making `suite_config` a class (maybe even a dataclass). That way it's more discoverable (e.g. autosuggestions in the IDE) and easier to document (`sphinx-autoapi` or similar).

Reply:
Done ✅ I had started to think about validation on the dict, but a dataclass is much better 🙂